lib/ASRouter.jsm

```diff
@@ -307,12 +319,33 @@ class _ASRouter {
     }
   }
 
+  _validPreviewEndpoint(url) {
+    try {
+      const endpoint = new URL(url);
```

Comment:
Should we also allow the SNIPPETS_ENDPOINT here since it's allowed for production?

Reply:
I whitelisted both production and preview:
https://github.com/piatra/activity-stream/blob/ef1f34f803703919e61e95ce55faa2c965e8d45c/lib/ASRouter.jsm#L24-L25

We could whitelist whatever SNIPPETS_ENDPOINT points to but is that necessary? If you change the pref value it will fetch the snippets because it goes through a different flow then preview.

Reply:
By SNIPPETS_ENDPOINT do you mean the exact URL or the hostname of the endpoint?

Reply:
oh right, ha! missed the production line somehow... :P

Reply:
Having the pref be allowed would let you work with a local provider more easily, although can also just use the admin panel for that so this probably fine